app/src/main/java/ca/carleton/gcrc/n2android_mobile1/connection/ConnectionSyncProcess.java

```diff
@@ -45,23 +65,88 @@ public void synchronize() throws Exception {
         try {
             Log.v(TAG, "Synchronization started");
 
-            List<JSONObject> remoteDocs = getRemoteSkeletonDocs();
+            List<JSONObject> remoteDocs = fetchAllDocuments();
+            updateLocalDocuments(remoteDocs);
+
+            updateRemoteDocuments(remoteDocs);
+
+        } catch(Exception e) {
+            throw new Exception("Error while synchronizing connection",e);
+        }
+    }
+
+    public List<JSONObject> fetchAllDocuments() throws Exception {
+        try {
+            List<JSONObject> remoteDocs = getDocsFromView("skeleton-docs");
+
+            Log.v(TAG, "Synchronization received " + remoteDocs.size() + " skeleton document(s)");
 
-            Log.v(TAG, "Synchronization received "+remoteDocs.size()+" skeleton document(s)");
+            // TODO: Let the user select which documents to sync.
+            // Update Documents from the Schema
+            Log.v(TAG, "Synchronization fetching subdocuments");
+
+            List<JSONObject> subdocuments = new ArrayList<>();
+            Set<String> subdocumentIdSet = new HashSet<String>();
+            Set<String> schemaIdSet = new HashSet<String>();
+
+            for(JSONObject skeletonDoc : remoteDocs) {
+                subdocumentIdSet.add(skeletonDoc.getString("_id"));
+            }
+
+            for (JSONObject doc : remoteDocs) {
+                if (doc.has("nunaliit_schema") && Objects.equals(doc.getString("nunaliit_schema"), "schema")) {
+                    schemaIdSet.add(doc.getString("name"));
+                }
+            }
+
+            List<JSONObject> subdocsForSchema = fetchDocumentsForSchemas(schemaIdSet);
+
+            for(JSONObject subdoc : subdocsForSchema) {
+                String id = subdoc.getString("_id");
+                if (!subdocumentIdSet.contains(id)) {
+                    subdocuments.add(subdoc);
+                    subdocumentIdSet.add(id);
```

Comment:
Just by curiosity: is there a reason we store the subdocuments in a List and their IDs in a Set instead of storing all the info in a Map? It seems like a common pattern in this class but i'm not sure why.